workers/unity/Packages/com.improbable.gdk.core/GameObjectRepresentation/SpatialOSBehaviourLibrary.cs

```diff
@@ -40,24 +41,29 @@ public SpatialOSBehaviourLibrary(ILogDispatcher logger)
             this.logger = logger;
         }
 
-        public void InjectAllReadersWriters(MonoBehaviour spatialOSBehaviour)
+        public Dictionary<uint, IReaderInternal> InjectAllReadersWriters(MonoBehaviour spatialOSBehaviour, Entity entity)
         {
             var spatialOSBehaviourType = spatialOSBehaviour.GetType();
             EnsureLoaded(spatialOSBehaviourType);
+            var createdReaderWriters = new Dictionary<uint, IReaderInternal>();
             if (invalidMonoBehaviourTypes.Contains(spatialOSBehaviourType))
             {
-                return;
+                return createdReaderWriters;
```

Comment:
what exactly will be stored in `invalidMonoBehaviourTypes`? so far it doesn't seem to be used

Reply:
It is used, types requiring multiple readers/writers (or both a reader and writer) of the same component get added when loaded (in `EnsureLoaded()`)